include/wpe/gamepad.h

```diff
@@ -142,7 +142,7 @@ struct wpe_gamepad_client_interface {
     void (*axis_changed)(void*, enum wpe_gamepad_axis, double);
 
     /*< private >*/
-    void (*_wpe_reserved1)(void);
+    void (*analog_button_changed)(void*, enum wpe_gamepad_button, double);
```

Comment:
Move `analog_button_changed` to line 143, right below `axis_changed`, and renumber the `_wpe_reservedX` so they start with `1`.

Reply:
I wonder if it would be possible to reuse `button_changed` for both kinds of buttons value, using double instead of boolean, without breaking the back compatibility.

Reply:
No, we cannot change the types of parameters or the number of parameters, because that would break both the ABI and the API. Adding a new function is fine, as done in this patch.